MissingSwitchDefault false positive on switch expressions when wrapped in return()

- [X] I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
- [X] I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
- [X] I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Test.java && echo "Success"
Success

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">

    <module name="TreeWalker">
        <module name="MissingSwitchDefault"/>
    </module>

</module>

$ cat Test.java
import java.util.Optional;

class Test {
    enum Case {
        One,
        Two,
    }

    Optional<String> faulty(Case value) {
        return (
            switch (value) {         // <- MissingSwitchDefault (line 11)
                case One -> Optional.of("One");
                case Two -> Optional.of("Two");
            }
        ).map(String::toUpperCase);
    }

    Optional<String> working(Case value) {
        var x = switch (value) {    // <- This one is fine
            case One -> Optional.of("One");
            case Two -> Optional.of("Two");
        };
        return x.map(String::toUpperCase);
    }
}

$ java "-Duser.language=en -Duser.country=US" -jar checkstyle-10.17.0-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /home/jdhalsey/IdeaProjects/test-http/src/main/java/Test.java:11:13: switch without "default" clause. [MissingSwitchDefault]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

Should not report a MissingSwitchDefault

---

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java
@@ -169,8 +169,17 @@ public class MissingSwitchDefaultCheck extends AbstractCheck {
      * @return true if part of a switch expression
      */
     private static boolean isSwitchExpression(DetailAST ast) {
-        return ast.getParent().getType() == TokenTypes.EXPR
-                || ast.getParent().getParent().getType() == TokenTypes.EXPR;
+        final int[] switchStatementParents = {
+            TokenTypes.SLIST,
+            TokenTypes.LITERAL_IF,
+            TokenTypes.LITERAL_ELSE,
+            TokenTypes.LITERAL_DO,
+            TokenTypes.LITERAL_WHILE,
+            TokenTypes.LITERAL_FOR,
+            TokenTypes.LABELED_STAT,
+        };
+
+        return !TokenUtil.isOfType(ast.getParent(), switchStatementParents);
     }
 
     /**

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheckTest.java
@@ -115,4 +115,15 @@ public class MissingSwitchDefaultCheckTest
                 expected);
     }
 
+    @Test
+    public void testMissingSwitchDefaultWithSwitchExpressionUnderMethodCall() throws Exception {
+        final String[] expected = {
+            "19:9: " + getCheckMessage(MSG_KEY),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputMissingSwitchDefaultCheckSwitchExpressionUnderMethodCall.java"),
+                expected);
+    }
+
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingswitchdefault/InputMissingSwitchDefaultCheckSwitchExpressionUnderMethodCall.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/missingswitchdefault/InputMissingSwitchDefaultCheckSwitchExpressionUnderMethodCall.java
@@ -1,0 +1,39 @@
+/*
+MissingSwitchDefault
+
+
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.coding.missingswitchdefault;
+
+import java.util.Optional;
+
+public class InputMissingSwitchDefaultCheckSwitchExpressionUnderMethodCall {
+    enum Case {
+        ONE,
+        TWO,
+    }
+
+    Optional<String> faulty(Case value) {
+        switch (value) {  // violation, 'switch without "default" clause.'
+            case ONE -> System.out.println("one");
+            case TWO -> System.out.println("two");
+        }
+
+        return (
+                switch (value) { // ok, switch expression must be exhaustive
+                    case ONE -> Optional.of("One");
+                    case TWO -> Optional.of("Two");
+                }
+        ).map(String::toUpperCase);
+    }
+
+    Optional<String> working(Case value) {
+        var x = switch (value) { // ok, switch expression must be exhaustive
+            case ONE -> Optional.of("One");
+            case TWO -> Optional.of("Two");
+        };
+        return x.map(String::toUpperCase);
+    }
+}

